core/templates/dev/head/domain/learner_answer_details/LearnerAnswerInfoBackendApiService.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to record learner answer info.
+ */
+
+require('domain/utilities/UrlInterpolationService.ts');
+
+require('domain/learner_answer_details/learner-answer-info.constants.ts');
+
+oppia.factory('LearnerAnswerInfoBackendApiService', [
+  '$http', 'UrlInterpolationService', 'SUBMIT_LEARNER_ANSWER_INFO_URL',
+  function($http, UrlInterpolationService, SUBMIT_LEARNER_ANSWER_INFO_URL) {
+    return {
```

Comment:
I think usually when we write these services, the private and public parts of the function are separated, can you maybe follow the same pattern?

Reply:
Done